src/examples/libpmemobj/hashmap/hashmap_atomic.c

```diff
@@ -112,6 +112,8 @@ create_hashmap(PMEMobjpool *pop, TOID(struct hashmap_atomic) hashmap,
 	size_t sz = sizeof(struct buckets) +
 			len * sizeof(struct entries_head);
 
+	pmemobj_persist(pop, D_RW(hashmap), sizeof(*D_RW(hashmap)));
```

Comment:
the persist below shouldn't be needed anymore.

Reply:
are needed to ensure order between fields above and allocation of buckets.

Reply:
how so? the correct persist placement was incorrect - we assumed that this entire function completed. Right now we are using the allocation as a "completion flag" for this function.

Reply:
I read your comment again. And yes, persist **below** is not needed :D

Reply:
Done.